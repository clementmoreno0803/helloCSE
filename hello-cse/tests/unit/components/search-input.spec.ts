import { describe, it, expect, vi } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import SearchInput from "@/components/search-input.vue"; // Assurez-vous que le chemin est correct

const mocksetSearchFilters = {
  setSearchFilters: vi.fn()
}

vi.mock('@/composables/UseMovie.ts', () => ({
  useMovie: () => mocksetSearchFilters
}))

const generateWrapper = () => {
  return shallowMount(SearchInput, {});
};

describe('SearchBar', () => {

  it('should render correctly', () => {
    const wrapper = generateWrapper();

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find('input').exists()).toBe(true);
  });

  it('should update inputSearch when user types', async () => {
    const wrapper = generateWrapper();
    const input = wrapper.find('input');
    await input.setValue('Inception');

    expect(wrapper.vm.inputSearch).toBe('Inception');
  });

  it('should change widthInput on focusin and focusout', async () => {
    const wrapper = generateWrapper();
    const input = wrapper.find('input');

    // Test de focusin
    await input.trigger('focusin');
    expect(wrapper.vm.widthInput).toBe('16rem');

    // Test de focusout
    await input.trigger('focusout');
    expect(wrapper.vm.widthInput).toBe('2rem');
  });

  it('should call setSearchFilters on keydown', async () => {
    const wrapper = generateWrapper();
    const input = wrapper.find('input');

    await input.setValue('Avatar');
    await input.trigger('keydown');

    expect(mocksetSearchFilters.setSearchFilters).toHaveBeenCalledWith('Avatar');
  });
});
