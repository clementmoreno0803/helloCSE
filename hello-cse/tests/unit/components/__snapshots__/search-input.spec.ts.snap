// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBar > should render correctly 1`] = `
<div
  class="search-bar"
  data-v-02b2df00=""
>
  <input
    class="search-input"
    data-testid="input-search"
    data-v-02b2df00=""
    placeholder="Veuillez renseigner un film"
    style="width: 2rem;"
    type="text"
  />
</div>
`;
